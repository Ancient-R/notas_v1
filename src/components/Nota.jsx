import React, { useState } from 'react';
import HeaderNota from '../components/HeaderNota'

const Nota = ({ nota, eliminaNota, setNuevoContenido }) => {
    
    // El nuevo contenido de la nota
    const [ contenido, setContenido] = useState(nota.contenido)
    const [ deshabilitado, setDeshabilitado ] = useState(true)

    const editarContenido = () =>{
        setDeshabilitado(false)
    }
    const actualizarContenido = () =>{
        setDeshabilitado(true)
        setNuevoContenido(contenido)
        nota.contenido = contenido
    }

    return ( 

        <div className="card mt-5 mb-3">
            { (nota.importancia === 'low') ?  
                <HeaderNota  
                    clase="alert-secondary"
                    titulo = {nota.titulo}>
                </HeaderNota>
                : 
                (nota.importancia === 'media') ?
                <HeaderNota  
                    clase="alert-warning"
                    titulo = {nota.titulo}>                
                </HeaderNota>
                :
                <HeaderNota  
                    clase="alert-danger"
                    titulo = {nota.titulo}>                
                </HeaderNota>
            }            
            <ul className="list-group list-group-flush">
                
                <textarea 
                    className="list-group-item editable"
                    disabled={deshabilitado}
                    value={contenido}
                    onChange={e => setContenido(e.target.value)}
                >{nota.contenido}</textarea>
            </ul>
            {/* Bot√≥n que edita la nota */}
            {deshabilitado ? <button
                className="btn alert-warning"
                onClick={ editarContenido }
            ><i className="fas fa-eraser"></i></button>:
            <button
                className="btn alert-success"
                onClick={ actualizarContenido }
            ><i className="far fa-check-circle"></i></button>}
            <button
                className="btn eliminar alert-danger"
                onClick={e=> eliminaNota(nota.id)}
            ><i className="far fa-trash-alt"></i></button>
        </div>
    );
}

export default Nota;