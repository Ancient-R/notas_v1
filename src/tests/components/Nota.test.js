import { shallow } from 'enzyme'
import React from 'react'
import Nota from '../../components/Nota'

const nota = {
    id: 1234,
    titulo: 'Título de testing',
    contenido: 'Contenido del título testing',
    importancia: 'low'
}
const eliminaNota = jest.fn();
const setNuevoContenido = jest.fn();

const wrapper = shallow(
    <Nota 
        nota={ nota }
        eliminaNota={ eliminaNota }
        setNuevoContenido={ setNuevoContenido }
    />
);

describe('Pruebas en el componente <Nota />', () => {
    test('debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe existir el componente <HeaderNota', () => {
        expect( wrapper.find('HeaderNota').exists() ).toBe( true );
    });
});