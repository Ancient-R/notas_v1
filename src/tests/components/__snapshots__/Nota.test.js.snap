// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Nota /> debe mostrarse correctamente 1`] = `
<div
  className="card mt-5 mb-3"
>
  <HeaderNota
    clase="alert-secondary"
    titulo="Título de testing"
  />
  <ul
    className="list-group list-group-flush"
  >
    <textarea
      className="list-group-item editable"
      disabled={true}
      onChange={[Function]}
      value="Contenido del título testing"
    >
      Contenido del título testing
    </textarea>
  </ul>
  <button
    className="btn alert-warning"
    onClick={[Function]}
  >
    <i
      className="fas fa-eraser"
    />
  </button>
  <button
    className="btn eliminar alert-danger"
    onClick={[Function]}
  >
    <i
      className="far fa-trash-alt"
    />
  </button>
</div>
`;
